### ISSUE TITLE
Microsoft.AspNetCore.Mvc.JsonOptions uses default instead of OC JOptions settings

### ISSUE THREAD

**Issue by wAsnk** [OrchardCMS/OrchardCore#16831]
### Describe the bug
Only `Microsoft.AspNetCore.Http.Json.JsonOptions` is [configured here](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs#L152) with the OC Json settings, while `Microsoft.AspNetCore.Mvc.JsonOptions` is kept as default, causing not minimal Apis using the default `Microsoft.AspNetCore.Mvc.JsonOptions` during model binding deserialization.

### Expected behavior
Both `JsonOption`s should use the same settings, a [similar configuration merge](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore.Abstractions/Json/JsonOptionsConfigurations.cs#L8) should be done for `Microsoft.AspNetCore.Mvc.JsonOptions` to ensure same behavior.

**Comment by hishamco** [comment]
@wAsnk feel free to submit a PR

**Comment by wAsnk** [comment]
> @wAsnk feel free to submit a PR

If I'll have spare time I'll do it and if someone else won't do it in the meantime.

**Comment by hishamco** [comment]
I will submit a PR tonight

**Comment by MikeAlhayek** [comment]
@wAsnk can you please share with us a content item that caused an exception in the minimalApi? 

It would be great if you have a repo to ensure it is properly fixed.

**Comment by wAsnk** [comment]
@MikeAlhayek 
Sorry it might've been misunderstandable. I was experiencing this with NOT minimal APIs, so with ApiControllers. In ApiControllers, `SystemTextJsonInputFormatter` and `SystemTextJsonOutputFormatter` is used during model binding, and the serializer there uses settings from `Microsoft.AspNetCore.Mvc.JsonOptions`.

I think setting globally the JsonOptions should be either for both or neither. But neither looks like the better solution.


Btw this wasn't my root problem, it was just a sidequest for me apparently.... My original problem was that I had a `Collection` with a private setter in a generated class, so while the OC defined settings where the `JsonObjectCreationHandling` was set to `Populate` the collection was populated. But with the default settings of `JsonOptions` the  `JsonObjectCreationHandling` is set to `Replace`, thus a private setter won't allow the collection to be set. So during deserialization I was missing data.
While I didn't know the exact problem I just copied the `Http.JsonOptions` to the `Mvc.JsonOptions` and used that everywhere.
My final solution now is that I removed the `Http.JsonOptions` like you [did in your PR](https://github.com/OrchardCMS/OrchardCore/pull/16837/files#diff-f29031364746f48326f6229655d9eb6a2b8d2dba01025a843fc90481198d000f). And also now I generate the class with a public setter for the collection. Now I can serialize-deserialize back and forth with the same result, unlike before.
